File: equipment-failure-prediction/backend/src/main/java/com/suyog/equipmentfailureprediction/service/AuthService.java
package com.suyog.equipmentfailureprediction.service;


import com.suyog.equipmentfailureprediction.dto.request.LoginRequest;
import com.suyog.equipmentfailureprediction.dto.request.RegisterRequest;
import com.suyog.equipmentfailureprediction.dto.response.AuthResponse;
import com.suyog.equipmentfailureprediction.dto.response.UserResponse;

public interface AuthService {

    UserResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: equipment-failure-prediction/frontend/src/services/authService.js
import api from './api';

export const authService = {
  login: async (email, password) => {
    const response = await api.post('/auth/login', { email, password });
    return response.data; // Response contains ApiResponse<AuthResponse> => { success, message, data: { token, tokenType, userId, fullName, email, role } }
  },

  register: async (fullName, email, password, role = 'ENGINEER') => {
    const response = await api.post('/auth/register', { fullName, email, password, role });
    return response.data; // ApiResponse<UserResponse>
  },
};

export default authService;
